src/main/java/nl/thehyve/ocdu/validators/patientDataChecks/SecondaryIdPatientDataCheck.java

```diff
@@ -0,0 +1,30 @@
+package nl.thehyve.ocdu.validators.patientDataChecks;
+
+import nl.thehyve.ocdu.models.OCEntities.Subject;
+import nl.thehyve.ocdu.models.OcDefinitions.MetaData;
+import nl.thehyve.ocdu.models.errors.ValidationErrorMessage;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Created by bo on 6/15/16.
+ */
+public class SecondaryIdPatientDataCheck implements PatientDataCheck{
+
+    @Override
+    public ValidationErrorMessage getCorrespondingError(int index, Subject subject, MetaData metaData) {
+
+        String ssid = subject.getSsid();
+        String commonMessage = getCommonErrorMessage(index, ssid);
+
+        ValidationErrorMessage error = null;
+        String secondaryId = subject.getSecondaryId();
+
+        if(!StringUtils.isBlank(secondaryId)) {
+            if(secondaryId.length() > 30) {
```

Comment:
It would be better create a static constant in this class (final static int MAX_SECONDARY_ID_LENGTH) and put this value there)